source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -730,8 +748,29 @@ namespace
                             {
                                 RtPrim sourceNode = source.getParent();
                                 InputPtr inputElem = valueElem->asA<Input>();
-                                inputElem->setNodeName(sourceNode.getName());
-                                inputElem->setOutputString(source.getName());
+                                if (sourceNode.hasApi<RtNodeGraph>())
+                                {
+                                    inputElem->setNodeGraphName(
+                                        sourceNode.getName());
+                                    inputElem->setOutputString(
+                                        source.getName());
+                                }
+                                else
+                                {
+                                    inputElem->setNodeName(sourceNode.getName());
+                                    RtNode rtSourceNode(sourceNode);
+                                    int numSourceNodeOutputs = 0;
+                                    auto outputIter = rtSourceNode.getOutputs();
+                                    while (!outputIter.isDone())
+                                    {
+                                        numSourceNodeOutputs++;
+                                        ++outputIter;
+                                    }
+                                    if (numSourceNodeOutputs > 1)
+                                    {
+                                        inputElem->setOutputString(source.getName());
```

Comment:
Only writes out output strings when the connected node is a multioutput

Reply:
Looks good. I've made a note to add in methods for getting counts so we don't have to iterate.